src/components/editgrid/EditGrid.js

```diff
@@ -263,7 +263,7 @@ export default class EditGridComponent extends NestedArrayComponent {
     const data = this.dataValue;
 
     (this.editRows || []).forEach((row, index) => {
-      const rowData = data[index];
+      const rowData = data[index] || {};
```

Comment:
This seems like it will detach this new row's data.  Why not something like this?

```
if (!data[index]) {
  this.dataValue[index] = {};
  data = this.dataValue;
}
```